### ISSUE TITLE
502 errors on certain Special pages

### ISSUE THREAD

**Issue by rjackson** [tfwiki/deployment#20]
LB or Varnish must be giving up waiting on MediaWiki for so long.

https://wiki.teamfortress.com/wiki/Special:WantedPages
https://wiki.teamfortress.com/wiki/Special:WantedFiles
https://wiki.teamfortress.com/wiki/Special:WantedTemplates

**Comment by rjackson** [comment]
With [updateSpecialPages.php](https://www.mediawiki.org/wiki/Manual:UpdateSpecialPages.php) we can move the heavy lifting from the page rendering process and into a background job. The script will generate the pages and jam them in a cache.

I'm doing a one-off run of that script now to:
* See what happens (does the frontend use the cached pages without any other changes?)
* See how long it takes

Shall report back shortly

**Comment by rjackson** [comment]
Fixed with

* tfwiki/deployment@44d693323405e073dfda3251b4038b3ca49a85da
* tfwiki/mediawiki@32848322f3b21e8dddf251adcccf35e1368e7d3f

**Comment by rjackson** [comment]
For reference:

```
Statistics                     [callback] completed in 0.02s
Ancientpages                   [QueryPage] got 1000 rows in 0.31s
BrokenRedirects                [QueryPage] got 362 rows in 0.40s
Deadendpages                   [QueryPage] got 38 rows in 0.61s
DoubleRedirects                [QueryPage] got 0 rows in 0.28s
FileDuplicateSearch            [QueryPage] cheap, skipped
ListDuplicatedFiles            [QueryPage] got 210 rows in 0.19s
LinkSearch                     [QueryPage] cheap, skipped
Listredirects                  [QueryPage] got 1000 rows in 0.06s
Lonelypages                    [QueryPage] got 5 rows in 0.75s
Longpages                      [QueryPage] cheap, skipped
MediaStatistics                [QueryPage] got 12 rows in 0.15s
MIMEsearch                     [QueryPage] got 0 rows in 0.00s
Mostcategories                 [QueryPage] got 1000 rows in 2.05s
Mostimages                     [QueryPage] got 1000 rows in 1.03s
Mostinterwikis                 [QueryPage] got 0 rows in 0.01s
Mostlinkedcategories           [QueryPage] cheap, skipped
Mostlinkedtemplates            [QueryPage] got 1000 rows in 13.72s
Mostlinked                     [QueryPage] got 1000 rows in 5m 50.00s
Mostrevisions                  [QueryPage] got 1000 rows in 5.09s
Fewestrevisions                [QueryPage] got 1000 rows in 4.97s
Shortpages                     [QueryPage] cheap, skipped
Uncategorizedcategories        [QueryPage] got 1 rows in 0.07s
Uncategorizedpages             [QueryPage] got 1 rows in 0.79s
Uncategorizedimages            [QueryPage] got 1 rows in 0.47s
Uncategorizedtemplates         [QueryPage] got 1000 rows in 0.06s
Unusedcategories               [QueryPage] got 76 rows in 0.06s
Unusedimages                   [QueryPage] got 1000 rows in 0.15s
Wantedcategories               [QueryPage] got 565 rows in 0.92s
Wantedfiles                    [QueryPage] got 1000 rows in 13.43s
Wantedpages                    [QueryPage] got 1000 rows in 3m 19.37s
Wantedtemplates                [QueryPage] got 397 rows in 2m 17.26s
Unwatchedpages                 [QueryPage] got 1000 rows in 0.08s
Unusedtemplates                [QueryPage] got 1000 rows in 0.10s
Withoutinterwiki               [QueryPage] got 1000 rows in 0.33s

```

**Comment by Tarkoviski** [comment]
It seems that Special:NewFiles still returns 502.

**Comment by rjackson** [comment]
@Tarkoviski Hmm, I'm not seeing that. Would you mind creating a new issue if it's still happening for you and I'll dig in deeper?

**Comment by Tarkoviski** [comment]
Hmm, I could create a new issue I think. I did some tests and noticed that when logged as "Tark" the page returns an error, but when in incognito/logged out/using a different account the page works fine.
